src/components/ToDoContext.tsx

```diff
@@ -0,0 +1,94 @@
+import React, {
+  useReducer,
+} from 'react';
+import { Todo } from '../types/Todo';
+
+export enum ACTIONS {
+  SET_LIST = 'setList',
+  SET_VISIBLE_LIST = 'setVisibleList',
+  SET_SEARCH_VALUE = 'setSearchValue',
+  SORT = 'sortBy',
+  SET_TODO = 'setTodo',
+}
+
+export enum FILTER {
+  SEARCH = 'Search',
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLITED = 'completed',
+}
+
+type Action = { type: ACTIONS.SET_LIST, payload: Todo[] }
+| { type: ACTIONS.SET_VISIBLE_LIST, payload: Todo[] }
+| { type: ACTIONS.SET_SEARCH_VALUE, payload: string }
+| { type: ACTIONS.SET_TODO, payload: Todo }
+| { type: ACTIONS.SORT, payload: string };
+
+interface State {
+  list: Todo[],
+  visibleList: Todo[],
+  sortBy: string,
```

Comment:
let's use enum here

Reply:
how exactly and what should I subsitute?